src/LinkContainer.js

```diff
@@ -66,42 +48,63 @@ class LinkContainer extends React.Component {
     }
 
     if (
-      target ||
-      event.defaultPrevented ||
-      isModifiedEvent(event) ||
-      !isLeftClickEvent(event)
+      !event.defaultPrevented && // onClick prevented default
+      event.button === 0 && // ignore right clicks
+      !this.props.target && // let browser handle "target=_blank" etc.
+      !isModifiedEvent(event) // ignore clicks with modifier keys
     ) {
-      return;
-    }
+      event.preventDefault();
 
-    event.preventDefault();
+      const { router } = this.context;
+      const { replace, to } = this.props;
 
-    this.context.router[action](
-      createLocationDescriptor(to, query, hash, state)
-    );
-  };
+      if (replace) {
+        router.replace(to);
+      } else {
+        router.push(to);
+      }
+    }
+  }
 
   render() {
-    const { router } = this.context;
-    const { onlyActiveOnIndex, to, children, ...props } = this.props;
-
-    props.onClick = this.onClick;
-
-    // Ignore if rendered outside Router context; simplifies unit testing.
-    if (router) {
-      props.href = router.createHref(to);
+    const {
+      children,
+      replace, // eslint-disable-line no-unused-vars
+      to,
+      exact,
+      strict,
+      activeClassName,
+      className,
+      activeStyle,
+      style,
+      isActive: getIsActive,
+      ...props,
+    } = this.props;
 
-      if (props.active == null) {
-        props.active = router.isActive(to, onlyActiveOnIndex);
-      }
-    }
+    const href = this.context.router.createHref(
+      typeof to === 'string' ? { pathname: to } : to
+    );
 
-    return React.cloneElement(React.Children.only(children), props);
+    return (
+      <Route
```

Comment:
@jquense We have no choice here - it's the way RRv4 works. I could've used [`matchPath`](https://github.com/ReactTraining/react-router/blob/7ce8ccfe0f38a1381f5456ee86c8a3358a1f1c78/packages/react-router/modules/matchPath.js) here but it'd be reinventing the wheel because `<Route>` already does the job.